spec-tests/src/test/java/org/candlepin/spec/activationkey/ActivationKeySpecTest.java

```diff
@@ -0,0 +1,651 @@
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        OwnerDTO anotherOwner = createOwner(adminClient);
+        ApiClient anotherUserClient = createUserClient(adminClient, anotherOwner);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+
+        assertNotFound(() -> anotherUserClient.activationKeys().deleteActivationKey(activationKey.getId()));
+    }
+
+    @Test
+    public void shouldAllowPoolsToBeAddedToAndRemovedFromActivationKeys() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        OwnerDTO anotherOwner = createOwner(adminClient);
+        ApiClient userClient = createUserClient(adminClient, owner);
+        ApiClient anotherUserClient = createUserClient(adminClient, anotherOwner);
+        ProductDTO product = createProduct(adminClient, owner);
+        PoolDTO pool = createPool(adminClient, owner, product);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+
+        adminClient.activationKeys().addPoolToKey(activationKey.getId(), pool.getId(), null);
+        ActivationKeyDTO updatedActivationKey = adminClient.activationKeys()
+            .getActivationKey(activationKey.getId());
+        assertEquals(1, updatedActivationKey.getPools().size());
+        adminClient.activationKeys().removePoolFromKey(activationKey.getId(), pool.getId());
+        updatedActivationKey = adminClient.activationKeys().getActivationKey(activationKey.getId());
+        assertEquals(0, updatedActivationKey.getPools().size());
+
+        userClient.activationKeys().addPoolToKey(activationKey.getId(), pool.getId(), null);
+        updatedActivationKey = userClient.activationKeys().getActivationKey(activationKey.getId());
+        assertEquals(1, updatedActivationKey.getPools().size());
+        userClient.activationKeys().removePoolFromKey(activationKey.getId(), pool.getId());
+        updatedActivationKey = userClient.activationKeys().getActivationKey(activationKey.getId());
+        assertEquals(0, updatedActivationKey.getPools().size());
+
+        assertNotFound(() -> {
+            anotherUserClient.activationKeys().addPoolToKey(activationKey.getId(), pool.getId(), null);
+        });
+    }
+
+    @Test
+    public void shouldAllowProductIdsToBeAddedToAndRemovedFromActivationKeys() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ProductDTO product = createProduct(adminClient, owner);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+
+        adminClient.activationKeys().addProductIdToKey(activationKey.getId(), product.getId());
+        ActivationKeyDTO updatedActivationKey = adminClient.activationKeys()
+            .getActivationKey(activationKey.getId());
+        assertEquals(1, updatedActivationKey.getProducts().size());
```

Comment:
We can also assert that the product was added to the updatedActivationKey

Reply:
done